liblangutil/SourceReferenceFormatter.cpp

```diff
@@ -164,10 +179,20 @@ void SourceReferenceFormatter::printSourceLocation(SourceReference const& _ref)
 void SourceReferenceFormatter::printExceptionInformation(SourceReferenceExtractor::Message const& _msg)
 {
 	// exception header line
-	errorColored() << _msg.severity;
-	if (m_withErrorIds && _msg.errorId.has_value())
-		errorColored() << " (" << _msg.errorId.value().error << ")";
-	messageColored() << ": " << _msg.primary.message << '\n';
+	if (_msg.errorId.has_value())
+	{
+		if (_msg.severity == "Error")
+			errorColored() << " (" << _msg.errorId.value().error << ")";
+
+		else if (_msg.severity == "Warning")
+			warningColored() << " (" << _msg.errorId.value().error << ")";
+
+		else if (_msg.severity == "Info")
+			infoColored() << " (" << _msg.errorId.value().error << ")";
+
+		else
+			unknownColored() << " (" << _msg.errorId.value().error << ")";
```

Comment:
You're not using your newly defined function here. Also, you should really use a `switch` now that we have it as an enum.

Reply:
Done ! 👍